src/librustc/hir/def.rs

```diff
@@ -71,7 +71,8 @@ pub enum Def<Id = hir::HirId> {
 
     // Value namespace
     Fn(DefId),
-    Const(DefId),
+    /// The `bool` refers to whether the constant is an anonymous constant (e.g. an array length)
+    Const(DefId, bool),
```

Comment:
Please don't add fields to `Def` variants, I'd rather you add a new variant (if at all).